worker_types/gwci-linux-beta/userdata

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+set -exv
+exec &> /var/log/userdata.log
+
+# Version numbers ####################
+LIVELOG_VERSION='v1.1.0'
+TASKCLUSTER_PROXY_VERSION='v5.1.0'
+######################################
+
+function retry {
+  set +e
+  local n=0
+  local max=10
+  while true; do
+    "$@" && break || {
+      if [[ $n -lt $max ]]; then
+        ((n++))
+        echo "Command failed" >&2
+        sleep_time=$((2 ** n))
+        echo "Sleeping $sleep_time seconds..." >&2
+        sleep $sleep_time
+        echo "Attempt $n/$max:" >&2
+      else
+        echo "Failed after $n attempts." >&2
+        exit 1
+      fi
+    }
+  done
+  set -e
+}
+
+start_time="$(date '+%s')"
+
+retry apt update
+DEBIAN_FRONTEND=noninteractive apt upgrade -yq
+retry apt install -y apt-transport-https ca-certificates curl software-properties-common git tar
+
+# install docker
+retry curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable"
+retry apt update
+apt-cache policy docker-ce | grep -qF download.docker.com
+retry apt install -y docker-ce
+sleep 5
+systemctl status docker | grep "Started Docker Application Container Engine"
+usermod -aG docker ubuntu
+
+# build generic-worker from bug1499054 branch
```

Comment:
Will we continue to build off this branch post-merge?

Reply:
This is just the staging environment, so I think it is ok to leave it in this state, as it reflects what is currently deployed in staging. When we deploy something else to staging, I think we'll update it at that time, which will keep the code in sync with what is deployed there.